tests/smoke/gmres.cpp

```diff
@@ -305,34 +431,28 @@ void grbProgram( const struct input &data_in, struct output &out ) {
 #endif
 		// read preconditioner P from file
 		if( !data_in.no_preconditioning ) {
-#ifdef DEBUG
-			std::cout << "Reading preconditioning matrix from "
-				<< data_in.precond_filename << " file \n";
-#endif
-			grb::utils::MatrixFileReader<
-				ScalarType,
-				std::conditional<
-					(sizeof( grb::config::RowIndexType ) > sizeof( grb::config::ColIndexType )),
-					grb::config::RowIndexType,
-					grb::config::ColIndexType
-				>::type
-			> parser_precond( data_in.precond_filename, data_in.direct );
-			if( parser_precond.m() != parser_precond.n() ) {
-				std::cerr << " matrix in " << data_in.precond_filename
-					<< " file, is not rectangular!";
-				rc = grb::ILLEGAL;
-			} else if( parser_precond.n() != n ) {
-				std::cerr << " Preconditioner P("
-					  << parser_precond.n() << ") mast have same dimensions as matrix A("
-					  << n << ") !\n";
-				rc = grb::ILLEGAL;
-			}
+			const auto &data = Preconditioner::getData().second;
 			rc = rc ? rc : buildMatrixUnique(
 				P,
-				parser_precond.begin( SEQUENTIAL ),
-				parser_precond.end( SEQUENTIAL ),
+				utils::makeNonzeroIterator<
+					grb::config::RowIndexType, grb::config::ColIndexType, ScalarType
+				>( data.cbegin() ),
+				utils::makeNonzeroIterator<
+					grb::config::RowIndexType, grb::config::ColIndexType, ScalarType
+				>( data.cend() ),
 				SEQUENTIAL
 			);
+			/* Once internal issue #342 is resolved this can be re-enabled
+			const RC rc = buildMatrixUnique(
+				P,
+				utils::makeNonzeroIterator<
+					grb::config::RowIndexType, grb::config::ColIndexType, ScalarType
+				>( data.cbegin() ),
+				utils::makeNonzeroIterator<
+					grb::config::RowIndexType, grb::config::ColIndexType, ScalarType
+				>( data.cend() ),
+				PARALLEL
+			);*/
```

Comment:
This code block should be in the directly preceding inner code block

Reply:
actually no, it is in the correct place -- it seems git diff was misleading here. Resolving without any action